examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,401 @@
+probability distribution of possible output values given an input
+(i.e., `maximum likelihood estimation <https://en.wikipedia.org/wiki/Likelihood_function>`_,
+a better approach is to perform the gradient descent in the *distribution space*, which is
+dimensionless and invariant with respect to the parametrization. As a result,
+each optimization step will always choose the optimum step-size for every
+parameter, regardless of the parametrization.
+
+In classical neural networks, the above process is known as
+*natural gradient descent*, and was first introduced by
+`Amari (1998) <https://www.mitpressjournals.org/doi/abs/10.1162/089976698300017746>`__.
+The standard gradient descent is modified as follows:
+
+.. math:: \theta_{t+1} = \theta_t - \eta F^{-1}\nabla \mathcal{L}(\theta),
+
+where :math:`F` is the `Fisher information matrix <https://en.wikipedia.org/wiki/Fisher_information#Matrix_form>`__.
+The Fisher information matrix acts as a metric tensor, transforming the
+steepest descent in the euclidean parameter space to the steepest descent in the
+distribution space.
+
+The quantum analog
+^^^^^^^^^^^^^^^^^^^^
+
+In a similar vein, it has been shown that the standard Euclidean geometry
+is sub-optimal for optimization of quantum variational algorithms
+`(Harrow and Napp, 2019) <https://arxiv.org/abs/1901.05374>`__.
+The space of quantum states instead possesses a unique invariant metric
+tensor known as the Fubini-Study metric tensor :math:`g_{ij}`, which can be used to
+construct a quantum analog to natural gradient descent:
+
+.. math:: \theta_{t+1} = \theta_t - \eta g^{+}(\theta_t)\nabla \mathcal{L}(\theta),
+
+where :math:`g^{+}` refers to the pseudo-inverse.
+
+.. note::
+
+    It can be shown that the Fubini-Study metric tensor reduces
+    to the Fisher information matrix in the classical limit.
+
+    Furthermore, in the limit where :math:`\eta\rightarrow 0`,
+    the dynamics of the system are equivalent to imaginary-time
+    evolution within the variational subspace, as proposed in
+    `McArdle et al. (2018) <https://arxiv.org/abs/1804.03023>`__.
+"""
+
+##############################################################################
+# Block-diagonal metric tensor
+# ----------------------------
+#
+# A block-diagonal approximation to the Fubini-Study metric tensor
+# of a variational quantum circuit can be evaluated on quantum hardware.
+#
+# Consider a variational quantum circuit
+#
+# .. math::
+#
+#     U(\mathbf{\theta})|\psi_0\rangle = V_L(\theta_L) W_L V_{L-1}(\theta_{L-1}) W_{L-1}
+#       \cdots V_{\ell}(\theta_{\ell}) W_{\ell} \cdots V_{0}(\theta_{0}) W_{0} |\psi_0\rangle
+#
+# where
+#
+# * :math:`|\psi_0\rangle` is the initial state,
+# * :math:`W_\ell` are layers of of non-parametrized quantum gates,
+# * :math:`V_\ell(\theta_\ell)` are layers of of parametrized quantum gates
+#   with :math:`n_\ell` parameters :math:`\theta_\ell = \{\theta^{(\ell)}_0, \dots, \theta^{(\ell)}_n\}`.

```

Comment:
```suggestion
#   with :math:`n` parameters :math:`\theta_\ell = \{\theta^{(\ell)}_0, \dots, \theta^{(\ell)}_n\}`.
```

Reply:
n is the number of parameters in each V_ell layer, right? so this does not need to have a subscript, no?

Reply:
or can different layers have different number of parameters?

Reply:
Different layers can have different number of parameters :slightly_smiling_face: